src/accelerate/state.py

```diff
@@ -193,3 +194,47 @@ def __repr__(self):
     @property
     def use_fp16(self):
         return self.mixed_precision != "no"
+
+    @staticmethod
+    def is_deepspeed_zero3_enabled():
+        if (
+            "deepspeed_plugin" in AcceleratorState._shared_state
+            and AcceleratorState._shared_state["deepspeed_plugin"] is not None
+        ):
+            deepspeed_plugin = AcceleratorState._shared_state["deepspeed_plugin"]
+            return deepspeed_plugin.hf_ds_config.is_zero3() and deepspeed_plugin.zero3_init_flag
+        return False
+
+    @staticmethod
+    def get_deepspeed_config():
+        ds_config = None
+        if (
+            "deepspeed_plugin" in AcceleratorState._shared_state
+            and AcceleratorState._shared_state["deepspeed_plugin"] is not None
+        ):
+            ds_config = deepcopy(AcceleratorState._shared_state["deepspeed_plugin"].hf_ds_config.config)
+            if ds_config["gradient_accumulation_steps"] == "auto":
+                ds_config["gradient_accumulation_steps"] = 1
+            if (
+                "train_micro_batch_size_per_gpu" not in ds_config
+                or ds_config["train_micro_batch_size_per_gpu"] == "auto"
+            ):
+                ds_config["train_micro_batch_size_per_gpu"] = 1
+            if ds_config["train_batch_size"] == "auto":
+                del ds_config["train_batch_size"]
+        return ds_config
+
+    @staticmethod
+    def set_deepspeed_config(hf_ds_config):
+        from .utils import DeepSpeedPlugin, HfDeepSpeedConfig
+
+        if isinstance(hf_ds_config, HfDeepSpeedConfig):
+            AcceleratorState._shared_state["deepspeed_plugin"] = DeepSpeedPlugin(
+                hf_ds_config=hf_ds_config, zero3_init_flag=True
+            )
+        else:
+            raise ValueError("`hf_ds_config` must be an instance of `accelerate.utils.HfDeepSpeedConfig`")
+
+    @staticmethod
+    def unset_deepspeed_config():
+        AcceleratorState._shared_state["deepspeed_plugin"] = None
```

Comment:
Let's check if the key exists before setting it a None?

Reply:
As long as you're sure it won't make any deepspeed related stuff break, fine by me.